extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/KafkaProcessor.java

```diff
@@ -561,14 +561,17 @@ public AdditionalBeanBuildItem kafkaClientBeans() {
     @BuildStep(onlyIf = IsDevelopment.class)
     @Record(ExecutionTime.RUNTIME_INIT)
     public void registerKafkaUiExecHandler(
+            Capabilities capabilities,
             BuildProducer<DevConsoleRouteBuildItem> routeProducer,
             KafkaUiRecorder recorder) {
-        routeProducer.produce(DevConsoleRouteBuildItem.builder()
-                .method("POST")
-                .handler(recorder.kafkaControlHandler())
-                .path(KAFKA_ADMIN_PATH)
-                .bodyHandlerRequired()
-                .build());
+        if (capabilities.isPresent("io.quarkus.vertx.http")) {
```

Comment:
There is a `Capability.VERTX_HTTP`

Reply:
Yes, but didn't find where/how it's exposed - there is nothing in the Vert.x HTTP extension.

@geoand any idea? Is it safe to provide and reuse this one, or was it intended for another purpose?

Reply:
Capabilities get "produced" by the configuration of the Maven plugin since they need to be present before the Quarkus build system determined the processing graph.

Reply:
Yes, but normally, it's declared in the extension descriptor, and it's not in this case.

Reply:
Ok, found it, it's in the maven plug-in configuration.

Reply:
All fixed!